pkg/controller/devicetainteviction/device_taint_eviction_test.go

```diff
@@ -1190,7 +1192,38 @@ func applyEventPair(tContext *testContext, event any) {
 	}
 }
 
-func startTestController(tCtx ktesting.TContext, informerFactory informers.SharedInformerFactory) *Controller {
+func newTestController(tCtx ktesting.TContext, clientSet *fake.Clientset) *Controller {
+	// fake.Clientset suffers from a race condition related to informers:
+	// it does not implement resource version support in its Watch
+	// implementation and instead assumes that watches are set up
+	// before further changes are made.
+	//
+	// If a test waits for caches to be synced and then immediately
+	// adds an object, that new object will never be seen by event handlers
+	// if the race goes wrong and the Watch call hadn't completed yet
+	// (can be triggered by adding a sleep before https://github.com/kubernetes/kubernetes/blob/b53b9fb5573323484af9a19cf3f5bfe80760abba/staging/src/k8s.io/client-go/tools/cache/reflector.go#L431).
```

Comment:
Should we fix it in client-go as well

Reply:
The fix in the fake package is likely to be complicated. Someone would have to implement proper resource version support for List and Watch. This is a nice research project, but IMHO not a high priority.

Reply:
Got it.

Reply:
I agree we should work around and fix this flaky test first.
In the meantime, just try to understand the root cause and create a pr at https://github.com/kubernetes/kubernetes/pull/131368/files#diff-b3ac8d4132005e736f9143b9140c44e789071d3ac5e8c89254e78a09391b57cfR170-R210 , do you think would it work

Reply:
No, I don't think that solves the problem.

Reply:
Noted, have closed it.